Packs/OpenSourceVulnerabilities/Integrations/OSV/OSV.py

```diff
@@ -0,0 +1,121 @@
+from CommonServerPython import *  # noqa: F401
+
+
+class Client(BaseClient):
+    def __init__(self, server_url, verify, proxy, headers, auth):
+        super().__init__(base_url=server_url, verify=verify, proxy=proxy, headers=headers, auth=auth)
+
+    def osv_get_vuln_by_id_new_request(self, id_):
+        headers = self._headers
+
+        response = self._http_request('get', f'v1new/vulns/{id_}', headers=headers)
+
+        return response
+
+    def osv_query_affected_by_commit_new_request(self, v1query_commit):
+        data = assign_params(commit=v1query_commit)
+        headers = self._headers
+
+        response = self._http_request('post', 'v1new/query', json_data=data, headers=headers)
+
+        return response
+
+    def osv_query_affected_by_package_new_request(self, v1query_version, v1query_package, v1query_ecosystem):
+        data = assign_params(version=v1query_version, package={'name': v1query_package, 'ecosystem': v1query_ecosystem})
+        headers = self._headers
+        response = self._http_request('post', 'v1new/query', json_data=data, headers=headers)
+
+        return response
+
+
+def osv_get_vuln_by_id_new_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    id_ = str(args.get('id_', ''))
+
+    response = client.osv_get_vuln_by_id_new_request(id_)
+    command_results = CommandResults(
+        outputs_prefix='OSV.Vulnerability',
+        outputs_key_field='',
+        outputs=response,
+        raw_response=response
+    )
+
+    return command_results
+
+
+def osv_query_affected_by_commit_new_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    v1query_commit = str(args.get('commit', ''))
+
+    response = client.osv_query_affected_by_commit_new_request(v1query_commit)
+    if response:
+        command_results = CommandResults(
+            outputs_prefix='OSV.VulnerabilityList',
+            outputs_key_field='',
+            outputs=response['vulns'],
+            raw_response=response
+        )
+    else:
+        return_error("Please check if the value provided is correct")
+    return command_results
+
+
+def osv_query_affected_by_package_new_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    v1query_version = str(args.get('version', ''))
+    v1query_package = str(args.get('packageName', ''))
+    v1query_ecosystem = str(args.get('ecosystem', ''))
+
+    response = client.osv_query_affected_by_package_new_request(v1query_version, v1query_package, v1query_ecosystem)
+    if response:
+        command_results = CommandResults(
+            outputs_prefix='OSV.VulnerabilityList',
+            outputs_key_field='',
+            outputs=response['vulns'],
+            raw_response=response
+        )
+    else:
+        return_error("Please check if the value provided is correct")
+    return command_results
+
+
+def test_module(client: Client) -> None:
+    try:
+        client.osv_get_vuln_by_id_new_request("OSV-2020-111")
```

Comment:
what if this given id does not exist? it fails the test, so is this id  essential for connection to the api, and without there's no connection?

Reply:
From the documentation, I couldn't find a generic way to ensure `200` response without giving an ID/commit_version/package_name. Maybe in the error/exception I can provide the suggestion to check if the `id` exists by using browser to go `https://api.osv.dev/v1/vulns/OSV-2020-111`? What do you suggest?

Reply:
its okay you can keep it as it is, no need to add something